src/dhttest/util/LocalStore.d

```diff
@@ -408,4 +409,61 @@ public struct NeoVerifier
             test!("==")(v, records[k]);
         }
     }
+
+    /***************************************************************************
+
+        Compares all record keys in the DHT channel against the keys in the
+        local store, using a DHT GetAll request in keys-only mode.
+
+        Params:
+            dht = DHT client to use to perform tests
+            channel = name of channel to compare against in DHT
+
+        Throws:
+            TestException upon verification failure
+
+    ***************************************************************************/
+
+    private void verifyGetAllKeysOnly ( DhtClient dht, cstring channel )
+    {
+        log.trace("\tVerifying channel with GetAll, keys-only");
+        auto task = Task.getThis();
+
+        bool[hash_t] keys;
+        bool duplicate;
+
+        void notifier ( DhtClient.Neo.GetAll.Notification info,
+            DhtClient.Neo.GetAll.Args args )
+        {
+            with ( info.Active ) switch ( info.active )
+            {
+                case received_key:
+                    if ( info.received_key.key in keys )
+                        duplicate = true;
+                    else
+                        keys[info.received_key.key] = true;
+                    break;
+
+                case started:
+                    break;
+
+                case finished:
+                    task.resume();
+                    break;
+
+                default:
+                    assert(false);
+            }
+        }
+
+        DhtClient.Neo.GetAll.Settings settings;
+        settings.keys_only = true;
+        dht.neo.getAll(channel.dup, &notifier, settings);
+        task.suspend();
+
+        test(!duplicate);
+        test!("==")(keys.length, this.local.data.length);
+        foreach ( k, v; this.local.data )
```

Comment:
I suggest to test `v.length == 0` as well.

Reply:
Hmm maybe I'm misunderstanding what you mean, but I think that's irrelevant.

A test case puts some records in the DHT and the same records in the local store. A bunch of verification functions run, including this one, which only checks the keys returned by a keys-only GetAll vs the keys in the local store. Why are the values in the local store relevant?

Reply:
I'm sorry, my bad. I read this as a `task-blocking` `GetAllKeys` request.